apollo-client-config-data/src/main/java/com/ctrip/framework/apollo/config/data/extension/webclient/customizer/spi/impl/ApolloClientWebClientAuthenticationCustomizerFactory.java

```diff
@@ -0,0 +1,187 @@
+import com.ctrip.framework.apollo.config.data.extension.properties.ApolloClientProperties;
+import com.ctrip.framework.apollo.config.data.extension.webclient.customizer.ApolloClientHttpBasicAuthenticationWebClientCustomizer;
+import com.ctrip.framework.apollo.config.data.extension.webclient.customizer.ApolloClientOauth2AuthenticationWebClientCustomizer;
+import com.ctrip.framework.apollo.config.data.extension.webclient.customizer.ApolloClientOauth2ReactiveAuthenticationWebClientCustomizer;
+import com.ctrip.framework.apollo.config.data.extension.webclient.customizer.spi.ApolloClientWebClientCustomizerFactory;
+import com.ctrip.framework.apollo.config.data.extension.webclient.filter.ApolloClientHttpBasicAuthenticationExchangeFilterFunction;
+import com.ctrip.framework.apollo.config.data.util.Slf4jLogMessageFormatter;
+import com.ctrip.framework.apollo.core.spi.Ordered;
+import org.apache.commons.logging.Log;
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties;
+import org.springframework.boot.context.properties.bind.BindHandler;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.boot.web.reactive.function.client.WebClientCustomizer;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction;
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction;
+
+/**
+ * @author vdisk <vdisk@foxmail.com>
+ */
+public class ApolloClientWebClientAuthenticationCustomizerFactory implements
+    ApolloClientWebClientCustomizerFactory {
+
+  private final ApolloClientPropertiesFactory apolloClientPropertiesFactory;
+
+  private final ApolloClientAuthorizedClientManagerFactory apolloClientAuthorizedClientManagerFactory;
+
+  private final ApolloClientReactiveAuthorizedClientManagerFactory apolloClientReactiveAuthorizedClientManagerFactory;
+
+  public ApolloClientWebClientAuthenticationCustomizerFactory() {
+    this.apolloClientPropertiesFactory = new ApolloClientPropertiesFactory();
+    this.apolloClientAuthorizedClientManagerFactory = new ApolloClientAuthorizedClientManagerFactory();
+    this.apolloClientReactiveAuthorizedClientManagerFactory = new ApolloClientReactiveAuthorizedClientManagerFactory();
+  }
+
+  @Override
+  public WebClientCustomizer createWebClientCustomizer(
+      ApolloClientProperties apolloClientProperties, Binder binder, BindHandler bindHandler,
+      Log log, ConfigurableBootstrapContext bootstrapContext) {
+    ApolloClientExtensionProperties extension = apolloClientProperties.getExtension();
```

Comment:
I'm not sure whether the authentication extension feature is common to other users since apollo itself doesn't support this function. And it seems to me this function adds a lot of complexities to this config-data sdk.

Reply:
remove the authentication extension